### ISSUE TITLE
[expo-notifications] Notification not working for IOS even though app received FCM message successfully

### ISSUE THREAD

**Issue by guanhuay** [expo/expo#7643]
## 🐛 Bug Report
I just tried integrate the newly release SDK 37 expo-notifications on my project. I have to says, this feature is awesome and very easy to setup.

However, I discover a bug with it. It is while when running app on IOS, I won't able to display any local notification on IOS, Android is working as expected .

Please refer to Step To Reproduce for code sample.

### Environment

  Expo CLI 3.17.11 environment info:
    System:
      OS: macOS Mojave 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.12.0 - /usr/local/bin/node
      Yarn: 1.21.1 - /usr/local/bin/yarn
      npm: 6.11.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 3.17.11

Running on bare workflow + expo unimodules
iOS target 9
Android target: min 21

### Steps to Reproduce

Basically, i setup Firebase FCM to receive message from firebase. When I sent message to the device with the given fcmToken, I was able to see the FCM payload by console log it.

However, I failed to display any notification for IOS, I have ensured I added the push notification, and remote notification capabilities to IOS project.
<img width="685" alt="Screen Shot 2020-04-03 at 16 55 46" src="https://user-images.githubusercontent.com/16933391/78413388-0c65d580-75cc-11ea-8a4c-dfb952ecdc07.png">

Here is the code snippet:

```
const getNotificationPermissionStatus = async () => {
  const existingStatus = await Notifications.getPermissionsAsync();
  return existingStatus.granted || existingStatus.ios
    ? existingStatus === 3
    : false;
};

useEffect(() => {
  const requestNotificationPermission = async () => {
    let existingStatus = getNotificationPermissionStatus();

    if (!existingStatus) {
      return await Notifications.requestPermissionsAsync({
        android: {},
        ios: {
          allowAlert: true,
          allowSound: true,
          allowAnnouncements: true,
        },
      });
    }
  };

  requestNotificationPermission().then(() => {});
}, []);

// Config Notifications Handler
Notifications.setNotificationHandler({
  handleNotification: async () => {
    return {
      shouldShowAlert: true,
      showPlaySound: true,
      priority: AndroidNotificationPriority.HIGH,
    };
  },
});
const showNotification = async (content, trigger) => {
  try {
    const identifier = await Notifications.scheduleNotificationAsync({
      content,
      trigger,
    });
    // clear notification
    await Notifications.dismissNotificationAsync(identifier);
  } catch (error) {
    console.log(error);
  }
};

// Incoming Message Handler
useEffect(() => {
  const unsubscribeOnMessage = messaging().onMessage(
    async (remoteMessage) => {
      const {
        notification: { title, body },
      } = remoteMessage;

      const content = {
        title: title,
        message: body,
        sound: true,
        priority: AndroidNotificationPriority.HIGH,
      };
      await showNotification(content, null);
    }
  );

  return unsubscribeOnMessage;
}, []);

```

Also my app's notification setting looks like below, everything is on:

<img width="394" alt="Screen Shot 2020-04-03 at 17 26 57" src="https://user-images.githubusercontent.com/16933391/78414318-69fc2100-75d0-11ea-97dc-bf588496dc04.png">

### Expected Behavior

Calling Notifications.scheduleNotificationAsync(....) should display notification on IOS app.

### Actual Behavior

No notification displayed even I was able to receive message payload from firebase server.

### Reproducible Demo

Going to skip this one for now.

**Comment by brentvatne** [comment]
Hello! `Notifications.scheduleNotificationAsync()` schedules a *local notification* and so it is unrelated to the "firebase server". 

I tried to re-create the issue that you mentioned and I could not. See https://github.com/brentvatne/local-notif-ios-example for a minimal example of an app with expo-notifications that displays a notification a second after you hit the button.

**Comment by guanhuay** [comment]
> Hello! `Notifications.scheduleNotificationAsync()` schedules a *local notification* and so it is unrelated to the "firebase server". 
> 
> I tried to re-create the issue that you mentioned and I could not. See https://github.com/brentvatne/local-notif-ios-example for a minimal example of an app with expo-notifications that displays a notification a second after you hit the button.

Sure I would take a look at your example, but my argument is i was able to receive payload from firebase, when i pass the content to Notifications.scheduleNotificationAsync,  nothing show up.

**Comment by yaroslavnikiforov** [comment]
@guanhuay , the problem can be because of calling `dismissNotificationAsync` right after `scheduleNotificationAsync`. It removes notification displayed in the notification tray (Notification Center).

**Comment by guanhuay** [comment]
> @guanhuay , the problem can be because of calling `dismissNotificationAsync` right after `scheduleNotificationAsync`. It removes notification displayed in the notification tray (Notification Center). 

Thanks for the comment, it's not related, this is a bug wiyh latest react naive firebase library.

**Comment by cHaLkdusT** [comment]
Hi @guanhuay, I'm not sure if this helps, as per Apple:

> If a notification arrives while your app is in the foreground, you can silence that notification or tell the system to continue to display the notification interface. The system silences notifications for foreground apps by *default*, delivering the notification’s data directly to your app. You can use the notification data to update your app’s interface directly. For example, if a new sports score arrived, you would just update that information in your interface.

**Comment by guanhuay** [comment]
> Hi @guanhuay, I'm not sure if this helps, but as per Apple:
> 
> > If a notification arrives while your app is in the foreground, you can silence that notification or tell the system to continue to display the notification interface. The system silences notifications for foreground apps by _default_, delivering the notification’s data directly to your app. You can use the notification data to update your app’s interface directly. For example, if a new sports score arrived, you would just update that information in your interface.

Hi there, 

This bug was caused by this defect from react-native-firebase and has been fixed by 6.4.1.alpha0.

See this bug for more details: [https://github.com/invertase/react-native-firebase/issues/3367#issue-589864365](https://github.com/invertase/react-native-firebase/issues/3367#issue-589864365)
